src/methods/truncated_fourier.jl

```diff
@@ -0,0 +1,141 @@
+export TFTS, TAAFT
+
+"""   
+    TFTS(fϵ::Real)
+
+A truncated Fourier transform surrogate[^Nakamura2006] (TFTS). These surrogates were 
+designed to deal with data with irregular fluctuations superimposed over long 
+term trends, and are generated by leaving a fraction of the lower frequencies untouched 
+when performing the phase shuffling step (as opposed to randomizing all frequencies,
+like in [`RandomFourier`](@ref) surrogates).
+
+The truncation parameter `fϵ` is the ratio of the high frequency domain to the 
+whole frequency domain. For example, if the number of frequencies is 2000 and frequencies between 1500 and 2000 are to be 
+randomized, then `fϵ = 500/2000 = 0.25`. The appropriate value of `fϵ` strongly 
+depends on the data and time series length, and must be manually determined[^Nakamura2006],
+for example by comparing periodograms for the time series and the surrogates.
+
+The TFTS surrogate can be used to test the null hypothesis that the signal is 
+a stationary linear system generated the irregular fluctuations part of the signal[^Nakamura2006].
+
+The by changing the sign of `fϵ`, the implementation here also allows for truncating 
+at the opposite end of the spectrum. Thus, negative values of `fϵ` will leave 
+the lower frequencies intact and shuffle the higher frequencies (the absolute value 
+of `fϵ` still refers to the fraction of high frequency domain to total frequency domain).
+
+[^Nakamura2006]: Nakamura, Tomomichi, Michael Small, and Yoshito Hirata. "Testing for nonlinearity in irregular fluctuations with long-term trends." Physical Review E 74.2 (2006): 026205.
+"""
+struct TFTS <: Surrogate
+    fϵ # ratio of high-frequency domain to the whole domain
+
+    function TFTS(fϵ::Real)
+        fϵ >= 0 || error("`fϵ` must be >= 0")
+        new(fϵ)
+    end
+end
+TFTS() = error("The truncation factor `fϵ` must be supplied")
```

Comment:
I don't think this line of code should be implemented (for source code sanity). Adding an error message for possible absence of arguments doesn't make sense: it is mandatory to read the documentation string to use any function, and the docstring tells you the arguments.